packages/stark-ui/src/modules/generic-search/classes/abstract-search-component.ts

```diff
@@ -150,7 +149,10 @@ export abstract class AbstractStarkSearchComponent<SearchResultsType, CriteriaTy
 	 * Invoke the genericSearchService.search() method and emits the results. If no searchCriteria object is passed, then the current
 	 * form's working copy is used.
 	 */
-	public performSearch(searchCriteria: CriteriaType = this.workingCopy): void {
+	public performSearch(searchCriteria: CriteriaType | undefined = this.workingCopy): void {
+		if (!searchCriteria) {
```

Comment:
Why can the `searchCriteria` be undefined? I don't think this if() block is needed either

Reply:
Good catch, I initialy did this because of `this.workingCopy` not having an initial value. But later I changed it to a definite assignment and forgot to revert this code.
Although I don't know if this entire package is safe to use in it's current state. Because `this.workingCopy` is assigned in a subscription. 🤔

Reply:
Well, the subscription is a pre-requisite for the GenericSearch component to work and it is documented so the developer is aware of it :wink: